lib/Ix/Context/WithAccount.pm

```diff
@@ -87,7 +97,12 @@ sub txn_do ($self, $code) {
     Carp::confess("We already have state before starting a transaction?!");
   }
 
-  # Start of a tree? Localize state so it goes away when we're done
+  # Start of a tree? Localize state and scratchpad so they go away when we're
```

Comment:
We're not actually localizing the scratchpad so this comment is wrong.

We could either fix the comment, or fix the code so we actually do localize the scratchpad (and then we don't have to clear it when done!)